Add information about preflight hardware checks (#501)
harvester/docs: docs/install/pxe-boot-install.md

:::info

Nodes need to have at least **8 GiB** of RAM because the installer loads the full ISO file into tmpfs.

> :::

Author: Sorry, just to clarify, do you mean the existing note about needing at least 8GiB RAM doesn't need to stand out? Or my new larger note about how Harvester runs the automated checks should just be a regular paragraph?

> :::

Author: OK, relevant preview pages are:
- https://65ddaa7cf030aba9b07816a5--harvester-preview.netlify.app/v1.3/install/index/
- https://65ddaa7cf030aba9b07816a5--harvester-preview.netlify.app/v1.3/install/pxe-boot-install

:::info

> During automated installation Harvester will perform the same inital checks as when booting via ISO, to ensure the system meets minimum hardware requirements for production use (i.e. bare metal with 16 CPU cores and 64GiB RAM). If any of these checks fail, installation will abort and warnings will be printed to the system console and saved to `/var/log/console.log` in the installation environment.

Reviewer: ```suggestion
During automated installation, Harvester will perform the same initial checks as when booting via ISO to ensure the system meets minimum

> During automated installation, Harvester will perform the same initial checks as when booting via ISO, to ensure the system meets minimum hardware requirements for production use (i.e. bare metal with 16 CPU cores, 64GiB RAM and 10Gbps ethernet). If any of these checks fail, installation will abort and warnings will be printed to the system console and saved to `/var/log/console.log` in the installation environment.
> 
> If you need to override this behavior, set the `harvester.install.skipchecks=true` kernel parameter. In this case, any warning messages will still be logged to `/var/log/console.log`, but the installation will proceed regardless of whether production hardware requirements are met.
> 
> :::

Reviewer 2: ```suggestion
:::info
The installer automatically checks if the hardware meets the [minimum requirements](https://docs.harvesterhci.io/v1.3/install/requirements/#hardware-requirements) for production use. If any of the checks fail, installation is stopped, and warnings are printed to the system console and saved to `/var/log/console.log` in the installation environment.

To override this behavior, set the kernel parameter `harvester.install.skipchecks=true`. When set to `true`, warning messages are still saved to `/var/log/console.log`, but the installation proceeds even if hardware requirements for production use are not met.
:::
```

Reviewer 2: I replaced the requirement examples with a link to the "Hardware Requirements" table to future-proof the content.

Failing to get IP from the DHCP server would cause iPXE booting to fail. You can add parameter `rd.net.dhcp.retry=<cnt>`
to retry DHCP request for `<cnt>` times.

> ### `harvester.install.skipchecks=true`
> 
> If Harvester's initial checks fail because the system does not meet the minimum hardware requirements for production use, installation will abort. To override this behaviour, set the `harvester.install.skipchecks=true` kernel parameter. In this case, any warning messages will be logged to `/var/log/console.log`, and the installation will proceed regardless of whether production hardware requirements are met.

Reviewer 2: ```suggestion
### `harvester.install.skipchecks=true`

Installation is stopped if the hardware checks fail (because the minimum requirements for production use are not met). To override this behavior, set the kernel parameter `harvester.install.skipchecks=true`. When set to `true`, warning messages are still saved to `/var/log/console.log`, but the installation proceeds even if hardware requirements for production use are not met.
```
